Explorer/Assets/DCL/Landscape/Systems/LandscapeCollidersCullingSystem.cs

```diff
@@ -12,40 +15,47 @@
 
 namespace DCL.Landscape.Systems
 {
-    [UpdateInGroup(typeof(RealmGroup))]
-    [UpdateAfter(typeof(UpdateCurrentSceneSystem))]
+    [UpdateInGroup(typeof(PostRenderingSystemGroup))]
+    [UpdateBefore(typeof(TeleportPositionCalculationSystem))]
     [LogCategory(ReportCategory.LANDSCAPE)]
     public partial class LandscapeCollidersCullingSystem : BaseUnityLoopSystem
     {
-        private readonly TerrainGenerator terrainGenerator;
+        private readonly TerrainGenerator terrain;
         private readonly IScenesCache sceneCache;
         private readonly ILoadingStatus loadingStatus;
         private readonly Entity playerEntity;
 
         private Vector2Int prevParcel = new (int.MaxValue, int.MaxValue);
 
-        public LandscapeCollidersCullingSystem(World world, TerrainGenerator terrainGenerator, IScenesCache sceneCache, ILoadingStatus loadingStatus) : base(world)
+        public LandscapeCollidersCullingSystem(World world, TerrainGenerator terrain, IScenesCache sceneCache, ILoadingStatus loadingStatus) : base(world)
         {
-            this.terrainGenerator = terrainGenerator;
+            this.terrain = terrain;
             this.sceneCache = sceneCache;
             this.loadingStatus = loadingStatus;
+
             playerEntity = world.CachePlayer();
         }
 
         protected override void Update(float t)
         {
-            if (!terrainGenerator.IsTerrainShown || loadingStatus.CurrentStage != LoadingStatus.LoadingStage.Completed) return;
-
-            Vector2Int newParcel = World.Get<CharacterTransform>(playerEntity).Transform.ParcelPosition();
-
-            if (prevParcel == newParcel)
+            // Enable terrain for proper raycasting on teleportation
+            ref PlayerTeleportIntent teleportIntent = ref World.TryGetRef<PlayerTeleportIntent>(playerEntity, out bool hasTeleportIntent);
```

Comment:
Query? Also, please cheack `TeleportUtils`. I used it on the new scene loading, I think you can use it as well

Reply:
since I have there fast `return`,  using query would make it more unreadable. And since I need to modify position, I need to use `ref`, so `TeleportUtils` are not applicable here